### ISSUE TITLE
windows starts 1024x768 resolution even SCREEN_RESOLUTION changed to 1920x1080x24

### ISSUE THREAD

**Issue by toni-moreno** [aerokube/windows-images#26]
Related also with https://github.com/aerokube/windows-images/issues/25

As you can see in this video . ( wait  20 seconds , to see vm running) 

[VIDEO](https://drive.google.com/file/d/11sPQYxd2xXJC8TaQi1BYeMwJqmZE_GA5/view?usp=sharing)

The container (X11) screen resolution is 1920x1080 but windows qemu vm is started at 1024x768. 

There is any way to sync VM resolution with container Resolution ?

**Comment by aandryashin** [comment]
Unfortunately resolution is windows setup and not qemu parameter...

вс, 24 нояб. 2019 г., 12:58 Toni Moreno <notifications@github.com>:

**Comment by emilorol** [comment]
@toni-moreno I solved this issue by setting the screen resolution higher before saving the snapshot. I installed the video driver that comes in the same package as the network card. I end up setting the screen resolution to 4K so that I don't need to keep resizing if the needs changes. The last thing I did was pass the screen resolution to the container so that I can see the whole windows screen.

My `broswers.json` looks like this:

```
    "internet explorer": {
        "default": "11.0",
        "versions": {
            "11.0": {
                "image": "nexus.internal.com:8443/docker/windows:browsers",
                "port": "4444",
                "path": "/",
                "privileged": true,
                "env": ["SCREEN_RESOLUTION=3200x2450x24"]
            }
        }
    },
    "MicrosoftEdge": {
        "default": "18.0",
        "versions": {
            "18.0": {
                "image": "nexus.internal.com:8443/docker/windows:browsers",
                "port": "5555",
                "path": "/",
                "privileged": true,
                "env": ["SCREEN_RESOLUTION=3200x2450x24"]
            }
        }
    },
```
The windows VM inside the container looks like this:

![windows-selenoid](https://user-images.githubusercontent.com/9518604/69972535-cbf82d00-14ef-11ea-975e-a36b77f90a19.png)

**Comment by vania-pooh** [comment]
There is even no command in Windows to set screen resolution.

**Comment by emilorol** [comment]
@vania-pooh you are correct, there is no command you can pass that will change windows resolution on the fly, that's why I am suggesting to change the windows resolution before saving the snapshot and then adjust the VNC viewer resolution to see all of your screen as need it.

**Comment by vania-pooh** [comment]
@emilorol yes, this seems to be the only possible solution right now.

**Comment by t880216t** [comment]
I found this tool nircmd http://www.nirsoft.net/utils/nircmd.html
In Windows, it can change the resolution setting through the command line.
Like this:

`> nircmd.exe setdisplay 1920 1080 32
`
But I don't know how to let qmeu call it with the specified parameters when it starts.

**Comment by vania-pooh** [comment]
@t880216t should be called during Windows startup. No way to call in qemu.

**Comment by t880216t** [comment]
> @t880216t should be called during Windows startup. No way to call in qemu.

I found temporary methods that can be solved dynamically.

1. Start a flask server to receive dynamic setting parameters.
2. Call the win32api library to set the resolution of Windows

Use it like this after the sessionin starts:
```
driver.get('http://127.0.0.1:5000/setDisplay?width=1920&height=1080')
driver.get('https://www.google.com')
```
This is a short flash server:
Dependent libraries:
```
# requirements.txt
click==8.1.3
colorama==0.4.5
Flask==2.2.2
itsdangerous==2.1.2
Jinja2==3.1.2
MarkupSafe==2.1.1
pywin32==304
Werkzeug==2.2.2
```
Source code:
``` python
# run.py
from flask import Flask
from flask import request
from flask import jsonify
import win32api

app = Flask(__name__)

def setScreen(width, height):
    dm = win32api.EnumDisplaySettings(None, 0)
    dm.PelsWidth = int(width)
    dm.PelsHeight = int(height)
    dm.BitsPerPel = 32
    dm.DisplayFixedOutput = 0
    win32api.ChangeDisplaySettings(dm, 0)

@app.route('/setDisplay', methods=['GET'])
def index():
    height = request.args.get('height')
    width = request.args.get('width')
    setScreen(width, height)

    return jsonify({'width': width, 'height': height})

if __name__ == '__main__':
    app.run(host='127.0.0.1', port=5000)

```
Keep the service running like as the browser webdriver server.
```
python run.py
```
